### ISSUE TITLE
Regression: /user/repos search cannot mix type and visibility/affiliation

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#1107]
There's two integration tests which are failing currently after merging #1096:

 - RepositoriesClientTests+TheGetAllForCurrentMethod.CanChangeSortDirection
 - RepositoriesClientTests+TheGetAllForCurrentMethod.CanSortResults

These both return an error:

```
HTTP/1.1 422 Unprocessable Entity
[other headers here]

{"message":"If you specify visibility or affiliation, you cannot specify type.","documentation_url":"https://developer.github.com/v3/repos/#list-your-repositories"}
```

And if you look at the URLs that we send:

```
GET https://api.github.com/user/repos?type=all&sort=created&direction=asc&visibility=public&affiliation=owner HTTP/1.1
[other headers here]
```

We just need to be smarter with the defaults so that they're not overlapping.

**Comment by AlexP11223** [comment]
What do you mean by "be smarter with the defaults"?
Do we need to add something like Default/NotSpecified (with null value) to Visibility and Affiliation enums?

**Comment by shiftkey** [comment]
Fixed in #1132
